core/unix/module_elf.c

```diff
@@ -405,6 +405,29 @@ module_walk_program_headers(app_pc base, size_t view_size, bool at_map, bool dyn
                 });
             }
         }
+        if (max_end + load_delta < base + view_size) {
+            /* i#3900: in-memory-only VDSO has a "loaded" portion not in a PT_LOAD
+             * official segment.  This confuses other code which takes the endpoint
+             * of the last segment as the endpoint of the mappings.  Our solution is
+             * to create a synthetic segment.
+             */
+            LOG(GLOBAL, LOG_INTERP | LOG_VMAREAS, 2,
+                "max_end " PFX " is smaller than map size " PFX ": assuming VDSO\n",
+                max_end + load_delta, base + view_size);
```

Comment:
Would be nice if the function would explain what each value is, I am for instance not really sure, but the entire module_elf is not extremely well documented what each function gets, does, and returns...

Reply:
Added more comments: it's all about finding the segments.